static_website/dispatch.ml

```diff
@@ -1,53 +1,50 @@
 open Lwt.Infix
-open V1_LWT
 
-module Main (C:CONSOLE) (FS:KV_RO) (S:Cohttp_lwt.Server) = struct
+let server_src = Logs.Src.create "server" ~doc:"HTTP server"
+module Server_log = (val Logs.src_log server_src : Logs.LOG)
 
-  let start c fs http =
+module Main (Clock:V1.CLOCK) (FS:V1_LWT.KV_RO) (S:Cohttp_lwt.Server) = struct
+  module Logs_reporter = Mirage_logs.Make(Clock)
 
-    let read_fs name =
-      FS.size fs name >>= function
+  let read_fs fs name =
+    FS.size fs name >>= function
+    | `Error (FS.Unknown_key _) -> Lwt.fail (Failure ("read " ^ name))
```

Comment:
I believe `Lwt.fail` will crash the unikernel but haven't tried for a while.
We should probably do something better, but this is all we've ever really done to date-- returning an error using `Rresult` or similar, for the caller to log and continue would seem sensible to me... (But I've not caught up on the v3 "break stuff" thread yet :)